frontend/src/routes/index/IndexPage.vue

```diff
@@ -18,15 +18,16 @@ import {
 } from '../../components/design-system/icons';
 import { mdiArrowLeft } from '@mdi/js';
 
-import IconButton, { ButtonSizeEnum } from '../../components/IconButton.vue';
 import { ref } from 'vue';
-const btn1Size = ref(ButtonSizeEnum.Medium);
-const btn2Size = ref(ButtonSizeEnum.Small);
-const incrementSize = (btnSize: number) => {
-  btnSize =
-    btnSize === Object.keys(ButtonSizeEnum).length / 2 - 1 ? 0 : btnSize + 1;
-  console.debug(btnSize);
-  return btnSize;
+import IconButton, {
+  type IconButtonProps,
+} from '@/components/design-system/input/IconButton.vue';
+const btn1Size = ref('medium' as IconButtonProps['size']);
```

Comment:
Pokud je to čistě pro ukázku, tak ok. Pokud ale se toto přetypování bude používat i nadále, tak jsem proti přetypování. Enum na toto funguje perfektně. Pokud je možnost se přetypování vyhnout, tak se mu vždy musí vyhnout. Má spoustu vedlejší nepříjemných efektů.

Reply:
Co je neprijemny efekt pretypovani v tomhle pripade?